WalletWasabi/CoinJoin/Client/Clients/CoinJoinClient.cs

```diff
@@ -584,8 +584,10 @@ private async Task TryRegisterCoinsAsync(ClientRound inputRegistrableRound)
 					// If our SatoshiClient does not yet know about the round, because of delay, then delay the round registration.
 					DelayedRoundRegistration = registration;
 				}
-
-				roundRegistered.Registration = registration;
+				else
+				{
+					roundRegistered.Registration = registration;
+				}
```

Comment:
I am not sure is this is correct or not, but if `roundRegistered` is `null` then we will run into a null exception.

Reply:
Yeah, you have discovered an NPE. The fix looks reasonable, but I guess only David or Lucas may know if it is really correct. Anyway, good catch.

Reply:
Good catch. It is hard to me to understand how could this be possible working without this. Only a complex interaction that guarantees that sooner or later the  `DelayedRoundRegistration` is assigned could explain it.